src/Mod/PartDesign/Gui/TaskRevolutionParameters.cpp

```diff
@@ -88,13 +88,14 @@ TaskRevolutionParameters::TaskRevolutionParameters(PartDesignGui::ViewProvider*
 
     setFocus ();
 
-    //show the parts coordinate system axis for selection
+    // show the parts coordinate system axis for selection
     PartDesign::Body * body = PartDesign::Body::findBodyOf ( vp->getObject () );
     if(body) {
         try {
             App::Origin *origin = body->getOrigin();
-            ViewProviderOrigin* vpOrigin;
-            vpOrigin = static_cast<ViewProviderOrigin*>(Gui::Application::Instance->getViewProvider(origin));
+            ViewProviderOrigin* vpOrigin = nullptr;

```

Comment:
vpOrigin should be directly initialized with
`ViewProviderOrigin* vpOrigin = static_cast<ViewProviderOrigin *>(...`

The dynamic_cast is not needed here because it won't happen that an App::Origin has a view provider different to ViewProviderOrigin

Reply:
done